src/app/containers/StoryPromo/index.jsx

```diff
@@ -1,25 +1,44 @@
 import React, { Fragment, useContext } from 'react';
-import { shape } from 'prop-types';
+import { shape, bool } from 'prop-types';
 import StoryPromoComponent, {
   Headline,
   Summary,
   Link,
 } from '@bbc/psammead-story-promo';
 import { storyItem } from '../../models/propTypes/storyItem';
-import StoryPromoFigure from './Figure';
+import FigureContainer from '../Figure';
 import Timestamp from '../Timestamp';
 import { ServiceContext } from '../../contexts/ServiceContext';
 import deepGet from '../../helpers/json/deepGet';
 
-const StoryPromo = ({ item }) => {
+const renderImage = (imageValues, lazyLoad) => {
+  if (!imageValues) {
+    return null;
+  }
+  const ratio = (imageValues.height / imageValues.width) * 100;
+  const src = `https://ichef.bbci.co.uk/news/660${imageValues.path}`;
+
+  return (
+    <FigureContainer
+      alt={imageValues.altText}
+      ratio={ratio}
+      src={src}
+      {...imageValues}
+      useFigure={false}
+      lazyLoad={lazyLoad}
+    />
+  );
+};
+
+const StoryPromo = ({ item, lazyLoadImage }) => {
   const { script } = useContext(ServiceContext);
   const headline = deepGet(['headlines', 'headline'], item);
   const url = deepGet(['locators', 'assetUri'], item);
   const summary = deepGet(['summary'], item);
   const timestamp = deepGet(['timestamp'], item);
   const imageValues = deepGet(['indexImage'], item);
 
-  const Image = imageValues && <StoryPromoFigure {...imageValues} />;
+  const Image = renderImage(imageValues, lazyLoadImage);
```

Comment:
Can you make this JSX please?

Reply:
@dr3 By making it jsx do you mean
```jsx
  let Image;
  if (imageValues) {
    const ratio = (imageValues.height / imageValues.width) * 100;
    const src = `https://ichef.bbci.co.uk/news/660${imageValues.path}`;
    Image = (
      <FigureContainer
        alt={imageValues.altText}
        ratio={ratio}
        src={src}
        {...imageValues}
        useFigure={false}
        lazyLoad={lazyLoadImage}
      />
    );
  }
```
inside `StoryPromo` or creating a `StoryPromoImage` react component instead of the `renderImage` function.

Reply:
No, soz I mean treat `renderImage` as a react component. Rather than `const Image = renderImage(imageValues, lazyLoadImage);`, do
```
const Image = (<ImageComponent imageValues={imageValues} lazyLoadImage={lazyLoadImage} />);
```